app/services/rag_wise/discussion_extraction_service.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+class RagWise::DiscussionExtractionService
+  def initialize(course, topic, posts)
+    @course = course
+    @topic = topic
+    @posts = Course::Discussion::Post.where(id: posts.pluck(:id))
+  end
+
+  def call
+    {
+      topic_title: sanitized_topic_title,
```

Comment:
Nit: Probably don't need the function `sanitized_topic_title` any longer.  
Can just set `sanitize_text(@topic[:title])` directly.

Reply:
Edited